projects/js-packages/connection/components/connect-user/index.jsx

```diff
@@ -0,0 +1,104 @@
+/**
+ * External dependencies
+ */
+import React, { useState, useEffect } from 'react';
+import { __ } from '@wordpress/i18n';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import InPlaceConnection from '../in-place-connection';
+import restApi from '../../tools/jetpack-rest-api-client';
+
+/**
+ * The user connection component.
+ *
+ * @param {object} props -- The properties.
+ * @param {Function} props.redirectFunc -- The redirect function (`window.location.assign()` by default).
+ * @param {string} props.connectUrl -- The authorization URL (no-iframe).
+ * @param {string} props.redirectUri -- The redirect admin URI.
+ * @param {string} props.inPlaceTitle -- The title for the In-Place Connection component.
+ * @param {boolean} props.forceCalypsoFlow -- Whether to go straight to Calypso flow, skipping the In-Place flow.
+ * @param {Function} props.onComplete -- The callback to be called when the connection is fully established.
+ * @param {string} props.from -- Where the connection request is coming from.
+ * @param {boolean} props.displayTOS -- Whether the site has connection owner connected.
+ *
+ * @returns {React.Component} The RNA connection component.
+ */
+const ConnectUser = props => {
+	const {
+		redirectFunc,
+		connectUrl,
+		redirectUri,
+		inPlaceTitle,
+		forceCalypsoFlow,
+		from,
+		onComplete,
+		displayTOS,
+	} = props;
+
+	const [ authorizationUrl, setAuthorizationUrl ] = useState( null );
+
+	if ( connectUrl && connectUrl !== authorizationUrl ) {
+		setAuthorizationUrl( connectUrl );
+	}
+
+	/**
+	 * Fetch the authorization URL on the first render.
+	 * To be only run once.
+	 */
+	useEffect( () => {
+		if ( ! authorizationUrl ) {
+			restApi
+				.fetchAuthorizationUrl( redirectUri )
+				.then( response => setAuthorizationUrl( response.authorizeUrl ) )
+				.catch( error => {
+					throw error;
+				} );
+		}
+	}, [] ); // eslint-disable-line react-hooks/exhaustive-deps
+
+	if ( ! authorizationUrl ) {
+		return null;
+		//throw new Error( 'Connect User URL (Authorization URL) is missing' );
+	}
+
+	if ( forceCalypsoFlow ) {
+		redirectFunc(
+			authorizationUrl +
+				( from
+					? ( authorizationUrl.includes( '?' ) ? '&' : '?' ) + 'from=' + encodeURIComponent( from )
+					: '' )
+		);
```

Comment:
Maybe in future iterations we could rely on `@wordpress/url` for things like this?
https://developer.wordpress.org/block-editor/reference-guides/packages/packages-url/#addQueryArgs

Reply:
Good idea, fixed!
https://github.com/Automattic/jetpack/pull/19837/commits/92fa0570bc1b3e32b0469a4aabda77bd0a4ef066